N24DCCN058/Bai tap buoi 4/Bai8.cpp

```diff
@@ -0,0 +1,22 @@
+#include <iostream>
+#include <cmath> 
+using namespace std;
+int main() {
+    float a , b , c;
+    cin >> a >> b >> c ;
+    if (abs(a-b) > c || abs(b-c) > a ){
```

Comment:
em chứng minh điều kiện này đc ko?:)
nếu nhập 1, 2, 3 thì bài em trả về tam giác vuông, nhưng 1 2 3 đâu phải là 1 tam giác
điều kiện tam giác: tổng 2 cạnh bất kì lớn hơn cạnh còn lại. (1 + 2) chỉ mới bằng 3 chứ ko lớn hơn 3 nên đây ko phải tam giác

Reply:
Trong 1 tam giác thì hiệu hai cạnh bất kỳ luôn luôn nhỏ hơn cạnh còn lại. 
Xét tam giác ABC thì luôn có :
|a - b| < c < a + b (1) 
|b-c| < a < b + c   (2) 
|c - a| < b < c + a  (3) 
Riêng 2 dấu trị tuyệt đối ở 2 sẽ bao gồm luôn trị tuyệt đối ở 3, nếu điều ngược lại như e ghi ở code thì in ra không là tam giác là đúng r ạ. Như ví dụ của anh thì 3 - 2 = 1 thì lúc này chương trình in ra màn hình là None r ạ

Reply:
> Trong 1 tam giác thì hiệu hai cạnh bất kỳ luôn luôn nhỏ hơn cạnh còn lại. Xét tam giác ABC thì luôn có : |a - b| < c < a + b (1) |b-c| < a < b + c (2) |c - a| < b < c + a (3) Riêng 2 dấu trị tuyệt đối ở 2 sẽ bao gồm luôn trị tuyệt đối ở 3, nếu điều ngược lại như e ghi ở code thì in ra không là tam giác là đúng r ạ. Như ví dụ của anh thì 3 - 2 = 1 thì lúc này chương trình in ra màn hình là None r ạ

![image](https://github.com/user-attachments/assets/16ab18dd-79d2-4181-8e66-44ade71c4ede)
hả:)

Reply:
Chứng minh em đúng r, viết code hơi sai tý thôi, anh thêm dấu >= vào chỗ abs là nó sẽ đúng , này là lỗi code thôi chứ anh kêu e chứng minh thì e mới viết ra thôi kk 💀

Reply:
Hiểu đơn giản là như này với mọi tam giác thì hiệu hai cạnh 1 tam giác luôn nhỏ hơn cạnh còn lại còn phủ định lại thì cho 3 điểm nếu hiệu 2 trong 3 điểm còn lại lớn hơn hoặc bằng cạnh còn lại thì đó không phải là một tam giác . Chỗ đó e thiếu dấu "=" :v

Reply:
Để trưa e update lại

Reply:
> Hiểu đơn giản là như này với mọi tam giác thì hiệu hai cạnh 1 tam giác luôn nhỏ hơn cạnh còn lại còn phủ định lại thì cho 3 điểm nếu hiệu 2 trong 3 điểm còn lại lớn hơn hoặc bằng cạnh còn lại thì đó không phải là một tam giác . Chỗ đó e thiếu dấu "=" :v

Lúc đầu a chưa cần biết e chứng minh sai hay đúng như anh test ra sai thì anh bảo sai thôi (sửa lại thì như e nói đó)
còn cách này là lạ nên anh kêu chứng minh thử cho vui thoi:) mục đích chính vẫn là sửa code vì nó bị sai